Linux - Workouts 1 - 3 Improvements (#2096)
enkidevs/curriculum: linux/user-and-file-management/manipulating-environment-variables/manipulating-environment-variables-iii.md

category: must-know

tags:
  - introduction

---

# Understanding `PATH`

---
## Content

Understanding the `PATH` environment variable and how to manipulate it is one of the most important things a beginner can master.  

The contents of `PATH` tell your shell where to look for executables.  It is therefore central to understanding what happens when you type a command in the terminal.

### What Does The `PATH` Variable Contain?

The `PATH` variable itself contains a `:`-delimited[1] list of directories:

```bash
echo $PATH
/usr/local/bin:/usr/bin:/bin
```

> > **Note**: the fact that `:`'s are used to delimit the directories here is a matter of *convention*.  Other parts of the system assume the list is `:`-delimited, but if in the 1970s we had settled on `,` or `;` instead, we would still be using that today.

Reviewer: another 💡moment :)

---
## Practice

> What environment variable is used by your shell to look for executables?
> 
> ???
> 
> * PATH
> * path
> * EXEC
> * exec

Reviewer: The answer is too obvious. How about we flip it? 

```suggestion
The `PATH` environment variable is used by your shell to look for ???.

* executables
* directories
* paths
* code

```
